pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -197,6 +198,45 @@ def setup_optimizers_in_pre_dispatch(self) -> bool:
         """
         return False
 
+    def restore_model_state_from_ckpt_path(self,
+                                           ckpt_path: str,
+                                           map_location=lambda storage, loc: storage) -> Tuple[Dict, bool]:
+        """
+        This function is used to load and restore the model state.
+
+        Args:
+            ckpt_path: Path to a checkpoint
+            map_location: lambda function to map checkpoint location
+
+        Return
+            checkpoint: Return loaded checkpoint
+            bool: Wether to load optimizer / lr_schedulers states from checkpoint
+
+        """
+        ckpt = pl_load(ckpt_path, map_location=map_location)
+        # restore datamodule states
+        if self.lightning_module.trainer.datamodule is not None:
+            self.lightning_module.trainer.datamodule.on_load_checkpoint(ckpt)
+
+        # hook: give user access to checkpoint if needed.
+        self.lightning_module.on_load_checkpoint(ckpt)
+        self.lightning_module.load_state_dict(ckpt['state_dict'])
+        return ckpt, True
+
+    def increment_accumulated_grad_global_step(self, trainer) -> None:
+        trainer.global_step += 1
+
+    @contextlib.contextmanager
+    def model_parallel_context(self) -> Generator:
```

Comment:
Merge has gone wrong here, this is already been merged into master as `model_sharded_context`. I think this should be removed, and we should refactor the API to use `model_sharded_context`

Reply:
Done !